src/Form/index.tsx

```diff
@@ -574,27 +574,30 @@ function Form({
   const getNewStep = async (newKey: any) => {
     // @ts-expect-error TS(2531): Object is possibly 'null'.
     let newStep = steps[newKey];
-    while (true) {
-      const loadCond = newStep.next_conditions.find((cond: any) => {
-        if (cond.element_type !== 'step') return false;
-        const notAuth =
-          cond.rules.find((r: any) => r.comparison === 'not_authenticated') &&
-          !initState.authId &&
-          // Re: firebaseConfirmationResult, the user hasn't authenticated yet
-          // but they're in the process of doing so and we don't want to
-          // consider them "unauthenticated" for the purposes of redirecting
-          !window.firebaseConfirmationResult;
-        const auth =
-          cond.rules.find((r: any) => r.comparison === 'authenticated') &&
-          initState.authId;
-        return notAuth || auth;
-      });
-      if (loadCond) {
-        if (changeStep(loadCond.next_step_key, newKey, steps, history)) {
-          clearLoaders(true);
-          return;
-        }
-      } else break;
+
+    const metadata =
+      integrations.stytch?.metadata ?? integrations.firebase?.metadata;
+    const authSteps = metadata?.auth_gate_steps ?? [];
+    if (authSteps.length > 0) {
+      const userAuthed = Boolean(initInfo().authId);
+      const nextStepIsProtected = authSteps.includes(newStep.id);
+      const findStep = (stepId: string) =>
+        // @ts-expect-error TS(2531): Object is possibly 'null'.
+        Object.values(steps).find((step) => step.id === stepId).key;
+      let nextStep: undefined | string;
+
+      if (userAuthed && !nextStepIsProtected)
+        nextStep = findStep(metadata.login_step);
```

Comment:
if login_step or logout_step happent o be invalid, empty string, etc. then the form will hard break here because 

Object.values(steps).find((step) => step.id === stepId).key;

will attempt to access the key prop on null. think we should check here to be robust, shoudl also fix the typescript error

Reply:
also i'm not sure we need this branch of the if statement. the user can be logged in and access a logged out step, the inverse is the only thing that we need to prevent.